src/chemnlp/utils.py

```diff
@@ -0,0 +1,12 @@
+from pathlib import Path
+from typing import Union
+
+import yaml
+
+
+def load_config(path: Union[str, Path]):
```

Comment:
I assume we will need `Union[str, Path]` more often. Shall we make `TypeAlias`es in a separate module? Besides, do we want to use `mypy` (or some other typechecker-I think in the current state it would still complain a lot)

Reply:
I think `TypeAlias` is from Python 3.10 onwards. Usually I'd just do things like:

```python
import jax.numpy as jnp
import numpy as np

Array = Union[np.ndarray, jnp.ndarray]
FloatList = Union[float, List[float]]
```

Re. `mypy` I think this is a bigger project question. We're in the (classic) half-adhering to typing (and I'm guilty here) place at the minute and most benefits will accrue if we enforce it project-wide.

Reply:
For `TypeAlias` I usually use the backport from https://pypi.org/project/typing-extensions/. But it is not that important and we can also do it later.